tensorflow/core/kernels/depthtospace_op.cc

```diff
@@ -112,7 +112,7 @@ class DepthToSpaceOp : public OpKernel {
     auto Tinput = input.tensor<T, kDims>();
     auto Toutput = outputs_tensor->tensor<T, kDims>();
 
-    if (std::is_same<Device, GPUDevice>::value) {
+    if constexpr (std::is_same<Device, GPUDevice>::value) {
```

Comment:
TF is still c++14 compatible, so `if constexpr` isn't available yet.

I don't know if that means the dummy template is still required.

Reply:
Ok, how about adding a preprocessor switch, like I did in my update?

Reply:
I'd rather not.  When we make the switch to c++17, we will likely do a large-scale change and using tooling to identify and update these expressions.  This would only get in the way of that.

Reply:
Ok, I'll then change this PR to just removing the dummy template that stopped working.